From c8ea8b5ba3f8a6a067fbd2490b3764eaf8bbc2a3 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 29 Aug 2019 14:13:50 +0200
Subject: [PATCH 3/5] also use NMake in codegen

---
 module/interface.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/module/interface.py b/module/interface.py
index 82686b0..6ba0a74 100644
--- a/module/interface.py
+++ b/module/interface.py
@@ -297,9 +297,9 @@ class OSQP(object):
         if project_type not in expectedProject:
             raise ValueError("Unknown value of 'project_type' argument.")
 
-        if project_type == 'Makefile':
+        if project_type in ('Makefile', ''):
             if system() == 'Windows':
-                project_type = 'MinGW Makefiles'
+                project_type = 'NMake Makefiles'
             elif system() == 'Linux' or system() == 'Darwin':
                 project_type = 'Unix Makefiles'
 
-- 
2.32.0.windows.2

