From a07ec3475fe4c6b68f360544d45ff43c0088bf7a Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 4 Jan 2023 22:05:46 +1100
Subject: [PATCH 4/5] do not rebuild libosqp here

---
 setup.py | 70 --------------------------------------------------------
 1 file changed, 70 deletions(-)

diff --git a/setup.py b/setup.py
index 4311726..a7dad01 100644
--- a/setup.py
+++ b/setup.py
@@ -86,9 +86,6 @@ cmake_args += ['-DPYTHON_INCLUDE_DIRS=%s' % sysconfig.get_python_inc()]
 
 # Define osqp and qdldl directories
 current_dir = os.getcwd()
-osqp_dir = os.path.join('osqp_sources')
-osqp_build_dir = os.path.join(osqp_dir, 'build')
-qdldl_dir = os.path.join(osqp_dir, 'lin_sys', 'direct', 'qdldl')
 
 
 # Interface files
@@ -136,73 +133,6 @@ if system() == 'Windows':
 # Add OSQP compiled library
 extra_objects = []
 
-'''
-Copy C sources for code generation
-'''
-
-# Create codegen directory
-osqp_codegen_sources_dir = os.path.join('src', 'osqp', 'codegen', 'sources')
-if os.path.exists(osqp_codegen_sources_dir):
-    sh.rmtree(osqp_codegen_sources_dir)
-os.makedirs(osqp_codegen_sources_dir)
-
-# OSQP C files
-cfiles = [os.path.join(osqp_dir, 'src', f)
-          for f in os.listdir(os.path.join(osqp_dir, 'src'))
-          if f.endswith('.c') and f not in ('cs.c', 'ctrlc.c', 'polish.c',
-                                            'lin_sys.c')]
-cfiles += [os.path.join(qdldl_dir, f)
-           for f in os.listdir(qdldl_dir)
-           if f.endswith('.c')]
-cfiles += [os.path.join(qdldl_dir, 'qdldl_sources', 'src', f)
-           for f in os.listdir(os.path.join(qdldl_dir, 'qdldl_sources',
-                                            'src'))]
-osqp_codegen_sources_c_dir = os.path.join(osqp_codegen_sources_dir, 'src')
-if os.path.exists(osqp_codegen_sources_c_dir):  # Create destination directory
-    sh.rmtree(osqp_codegen_sources_c_dir)
-os.makedirs(osqp_codegen_sources_c_dir)
-for f in cfiles:  # Copy C files
-    copy(f, osqp_codegen_sources_c_dir)
-
-# List with OSQP H files
-hfiles = [os.path.join(osqp_dir, 'include', f)
-          for f in os.listdir(os.path.join(osqp_dir, 'include'))
-          if f.endswith('.h') and f not in ('qdldl_types.h',
-                                            'osqp_configure.h',
-                                            'cs.h', 'ctrlc.h', 'polish.h',
-                                            'lin_sys.h')]
-hfiles += [os.path.join(qdldl_dir, f)
-           for f in os.listdir(qdldl_dir)
-           if f.endswith('.h')]
-hfiles += [os.path.join(qdldl_dir, 'qdldl_sources', 'include', f)
-           for f in os.listdir(os.path.join(qdldl_dir, 'qdldl_sources',
-                                            'include'))
-           if f.endswith('.h')]
-osqp_codegen_sources_h_dir = os.path.join(osqp_codegen_sources_dir, 'include')
-if os.path.exists(osqp_codegen_sources_h_dir):  # Create destination directory
-    sh.rmtree(osqp_codegen_sources_h_dir)
-os.makedirs(osqp_codegen_sources_h_dir)
-for f in hfiles:  # Copy header files
-    copy(f, osqp_codegen_sources_h_dir)
-
-# List with OSQP configure files
-configure_files = [os.path.join(osqp_dir, 'configure', 'osqp_configure.h.in'),
-                   os.path.join(qdldl_dir, 'qdldl_sources', 'configure',
-                                'qdldl_types.h.in')]
-osqp_codegen_sources_configure_dir = os.path.join(osqp_codegen_sources_dir,
-                                                  'configure')
-if os.path.exists(osqp_codegen_sources_configure_dir):
-    sh.rmtree(osqp_codegen_sources_configure_dir)
-os.makedirs(osqp_codegen_sources_configure_dir)
-for f in configure_files:  # Copy configure files
-    copy(f, osqp_codegen_sources_configure_dir)
-
-# Copy cmake files
-copy(os.path.join(osqp_dir, 'src',     'CMakeLists.txt'),
-     osqp_codegen_sources_c_dir)
-copy(os.path.join(osqp_dir, 'include', 'CMakeLists.txt'),
-     osqp_codegen_sources_h_dir)
-
 
 class build_ext_osqp(build_ext):
     def build_extensions(self):
