From 369b1d937afc373f940c0593263413e00fea1823 Mon Sep 17 00:00:00 2001
From: Jason Furmanek <furmanek@us.ibm.com>
Date: Thu, 12 Sep 2019 19:21:41 +0000
Subject: [PATCH 4/4] Disable MKL Tests on nonx86

---
 module/tests/mkl_pardiso_test.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/module/tests/mkl_pardiso_test.py b/module/tests/mkl_pardiso_test.py
index 72ba3dd..b8b7d93 100644
--- a/module/tests/mkl_pardiso_test.py
+++ b/module/tests/mkl_pardiso_test.py
@@ -20,6 +20,7 @@ class mkl_pardiso_tests(unittest.TestCase):
         self.l = np.array([0.0, 0.0])
         self.u = np.array([100.0, 100.0])
 
+    @unittest.skip("MKL Tests are only relevant on x86_64 systems")
     def test_issue14(self):
 
         m = osqp.OSQP()
-- 
2.21.0.windows.1

