From 0aa11a1db1300fb9a504c575d22ef389ae69ad46 Mon Sep 17 00:00:00 2001
From: Jason Furmanek <furmanek@us.ibm.com>
Date: Thu, 12 Sep 2019 19:21:41 +0000
Subject: [PATCH 4/5] Disable MKL Tests on nonx86

---
 module/tests/mkl_pardiso_test.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/module/tests/mkl_pardiso_test.py b/module/tests/mkl_pardiso_test.py
index b10a417..1e59d71 100644
--- a/module/tests/mkl_pardiso_test.py
+++ b/module/tests/mkl_pardiso_test.py
@@ -19,6 +19,7 @@ class mkl_pardiso_tests(unittest.TestCase):
         self.l = np.array([0.0, 0.0])
         self.u = np.array([100.0, 100.0])
 
+    @unittest.skip("MKL Tests are only relevant on x86_64 systems")
     def test_issue14(self):
 
         m = osqp.OSQP()
-- 
2.32.0.windows.2

